sopel/web.py

```diff
@@ -237,3 +237,39 @@ def iri_to_uri(iri):
     urlencode = urllib.urlencode
 else:
     urlencode = urllib.parse.urlencode
+
+
+def search_urls(text, exclusion_char=None, clean=False):
+    def trim_url(url):
```

Comment:
I think this should either become its own utility function (in `sopel.web` or `sopel.tools`), or become inlined code in the only place where it's used. Defining a function inside another is one of those oddball things that we should probably avoid doing unless there's a good reason.

Reply:
I moved this function mostly "as-is", adding the `exclusion_char` parameter.

Fun fact: since `find_url` was never call with `clean=True`, `trim_url` was (and still is) never called, ever. And I've no idea why.

Reply:
> Fun fact: since `find_url` was never call with `clean=True`, `trim_url` was (and still is) never called, ever. And I've no idea why.

I know why. It's because, before the flood of collaborators Sopel gained in the last few months, I left a small but very important change out of #1413 and there was nobody else looking at the code to call me out on my mistake. >.> So that'll be a good (bugfix) bugfix to add to the 6.6.x pile.

For this PR, you'll want to add a similar bugfix in the same place. I haven't written the patch just yet, but it's probably going to involve just passing `clean=True` from `title_auto` in the `url` module, so you can tentatively do the equivalent here.

Reply:
Update: The relevant bugfix is in #1515.

Reply:
>  so you can tentatively do the equivalent here

I'll add some tests and do this.

Reply:
I did extract it, I let you see if you agree with that.